### ISSUE TITLE
[CRUD] Fix accessibility of rows in Crud component

### ISSUE THREAD

**Issue by rkristelijn** [mui/toolpad#4825]
### What's the problem?

Dear Toolpad team, 

First of all thanks on this amazing work with MUI. It makes working with MUI so much simpler. I just discovered the Crud component and I'm amazed at the simplicity to set it up. 

Right from the bat I see that the rowclick is used to navigate from the list-view to the detail-view. I want to share some concerns here;

1. This blocks any inline edit that you may want to do in the list-view because the click navigates away
2. This is very hard to make accessible, the keyboard navigation doesn't work and the screenreader is unable to make out there is a way to navigate from the rowclick to the detail view
3. Multiselect is unusable as the first click navigates away

### What are the requirements?

Instead i would like to suggest to make on of the columns of type hyperlink, to create a so-called drilldown. This is easy for screen readers to make out there is a link to the detail view, multiselect is enabled, and you can have drilldowns to other components as well if you want

### Proposed solution

Imagine you have two 'business components': Contact and Account. A Contact can belong to only one account.

The Contact Crud:

| Id | Contact Name | Account |
|--------|--------|-------|
| drilldown to contact 1 | just some text | drilldown to account 23 |
| drilldown to contact 2 | just some text | drilldown to account 345 |

The Account Crud:

| Id | Account Name | Primary Contact |
|--------|--------|-------|
| drilldown to account 1 | just some text | drilldown to contact 456 |
| drilldown to account 2 | just some text | drilldown to contact 1 |

**Comment by apedroferreira** [comment]
Hi, I've considered your proposal by starting with an automatic "ID" cell that works as a link in https://github.com/mui/toolpad/pull/4943, but after discussing with other team members the final decision was to prioritize UX by making it possible to navigate to the item detail page by clicking the whole row.

Your concerns should still be addressable though!
1. You should be able to use `slotProps.list.dataGrid` in the `Crud` component to override any DataGrid props, as well as, for example, the `renderCell` property in any of your CRUD fields. We also plan to provide specific support for inline editing in the future.
3. Also should be possible to get around by overriding some props with `slotProps.list.dataGrid`. We also plan to support selecting and deleting multiple rows in the future, for example, as right now that would probably be difficult to do as an override to this component.

As for the accessibility/UX trade-off, the MUI X Data Grid currently doesn't allow us to fully satisfy both when choosing to make a whole row clickable to a different page... Maybe this could be improved/solved as an issue on their end?
